services/system/Accounts/ui/src/AccountSelection.tsx

```diff
@@ -197,13 +141,14 @@ export const AccountSelection = () => {
     return () => subscription.unsubscribe();
   }, [form]);
 
-  const [activeSearch, setActiveSearch] = useState("");
+  const [accountsToRender, setAccountsToRender]: [
+    AccountType[],
+    (accounts: AccountType[]) => void,
+  ] = useState<AccountType[]>([]);
```

Comment:
Usually this would just be:

`const [accountsToRender, setAccountsToRender] = useState<AccountType[]>([]);`

The rest is inferred and typed properly.

Reply:
Another comment about this: You're duplicating accounts state here. Fetching data (like accounts) and stuffing it into state and then managing that state as a new list of accounts is an antipattern. Better to do what was done in the previous version: fetch the accounts, track the state of the active search, and filter the list of accounts-to-render from the full list of accounts on render.